STYLE.md

```diff
@@ -315,3 +315,22 @@ you'll need to use an asynchronous-specific synchronization primitives, namely t
 itself. The documentation on `tokio`'s own
 [`Mutex`](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html), for example, calls out the
 specifics of when and where you might need to use it vs the one from `std::sync`.
+
+
+## New Configuration Fields vs CLI flags
+
+Vector makes the distinction between configuration items that are essential to understand data
+pipelines and runtime flags that determine the details of the runtime behavior. The main configuration
+generally lives in a `vector.toml` file in the current directory or at `/etc/vector/vector.toml`.
```

Comment:
Given we've expressed interest in pivoting to YAML as the primary configuration language, it may be worth generalizing this to avoid an additional place we'd need to update down the line.

Reply:
Great point! It might be a bit more confusing since all of the getting started docs refer to `vector.toml`, but it will be more correct and future-proof :)

[48146cc](https://github.com/vectordotdev/vector/pull/17784/commits/48146cc43c4973a492451214a95e54e5b6e7844b)

I am still referring to the configuration as a collection of fields which can be included in a file, which seems a bit pedantic / confusing as a definition. Let me know if you have better language to talk about "real" or "bona-fide" configurations as opposed to "runtime flags."